cppForSwig/ThreadSafeClasses.h

```diff
@@ -438,10 +438,9 @@ template<typename T, typename U> class TransactionalMap
       if (updatemap.size() == 0)
          return;
 
-      auto newMap = make_shared<map<T, U>>();
+      auto newMap = make_shared<map<T, U>>( move(updatemap));
 
       unique_lock<mutex> lock(mu_);
```

Comment:
This is a simple "thread safe" map container. When updates are submitted, the old map is copied, the new data set merged in, then the new map overwrites the old map.

All read/write operations on the old map have to happen after the lock, therefor this is correct.